### PR TITLE
Moving average for statistics in observers

### PR DESCRIPTION
**Pull request by raghuramank100**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26627 Default observer and fake-quant for backends
* **#26626 Moving average for statistics in observers**
* #26625 Emulate weight and activation only quant with fake quant, numerics test
* #26624 Quantization aware training: Freeze batch norm support
* #26623 Per channel fake quant
* #26516 Per-channel baseline
* #26520 control of observer/fake-quant operations
* #26622 Default histogram observer
* #26519 Serialization and range reduction support for Fake Quant/Observer
* #26420 Fake quantization enhancements for QAT/PTQ support
* #26518 Handle DeQuantStub() for QAT

Differential Revision: [D17520344](https://our.internmc.facebook.com/intern/diff/D17520344/)

### CODE REVIEW

File: torch/quantization/observer.py [hunk 1]
```diff
@@ -199,6 +199,7 @@ def __init__(self, **kwargs):
         #  aten/src/ATen/native/quantized/cpu/qconv.cpp
         #  This is not the optimal choice for non x86 backends as
         #  lose a bit of precision for activations.
+        self.averaging_constant = kwargs.pop('averaging_constant', 0.01)
```
Comment (dzhulgakov) [inline torch/quantization/observer.py]: nit: you don't need to do `pop`, just do `__init__(self, averaging_constant=0.01, **kwargs)` above

File: torch/quantization/observer.py [hunk 2]
```diff
@@ -327,6 +334,7 @@ class HistogramObserver(ObserverBase):
 
     def __init__(self, bins=2048, **kwargs):
         # bins: The number of bins used for histogram calculation.
+        self.averaging_constant = kwargs.pop('averaging_constant', 0.01)
```
Comment (dzhulgakov) [inline torch/quantization/observer.py]: same
